app/scripts/metamask-controller.js

```diff
@@ -2801,14 +2848,9 @@ export default class MetamaskController extends EventEmitter {
 
       // network management
       setProviderType: (type) => {
-        // when using this format, type happens to be the same as the networkClientId...
-        this.selectedNetworkController.setNetworkClientIdForMetamask(type);
         return this.networkController.setProviderType(type);
       },
       setActiveNetwork: (networkConfigurationId) => {
-        this.selectedNetworkController.setNetworkClientIdForMetamask(
-          networkConfigurationId,
-        );
         return this.networkController.setActiveNetwork(networkConfigurationId);
```

Comment:
this is now handled internally in the `SelectedNetworkController`: https://github.com/MetaMask/core/pull/3908/files#diff-5e705302d595033f24cac5f20d14ecf307709fd61c46a6d01bf9519bf5b91d86R154-R161